src/org/omegat/gui/theme/DelegatingLookAndFeel.java

```diff
@@ -62,8 +62,12 @@ protected static ImageIcon getIcon(String iconName) {
     protected final LookAndFeel systemLookAndFeel;
 
     public DelegatingLookAndFeel() throws Exception {
-        systemLookAndFeel = (LookAndFeel) Class.forName(UIManager.getSystemLookAndFeelClassName())
-                .getDeclaredConstructor().newInstance();
+        // Get System LaF instance.
+        // Calling Class.forName(UIManager.getSystemLookAndFeelClassName()).newInstance()
+        // on Java16 throw error "java.desktop does not export com.sun.java.swing.plaf.gtk"
+        // on Linux, and for aqua LaF on macOS. (Bug#1052)
+        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
```

Comment:
I like that we're not hard-coding the class name, but this now has the side effect of changing the active LAF.

It looks like there is an API in Java 9+ to instantiate a LAF without installing it:
http://hg.openjdk.java.net/jdk9/jdk9/jdk/rev/174b223645b7#l1.32
https://docs.oracle.com/javase/9/docs/api/javax/swing/UIManager.html#createLookAndFeel-java.lang.String-

Is it impossible to call this API via reflection?

In [DesktopSupport](https://github.com/amake/DesktopSupport) I do some very tricky things to bridge a Java 8 and Java 9+ API; I suppose that would be possible here but very complicated to do.

Reply:
Actually I think this API is a good candidate to be added to DesktopSupport.

Reply:
I just added a wrapper for `UIManager#createLookAndFeel` in DesktopSupport, but I haven't tested it yet. If you wanted to try you could build and install it locally.

Reply:
Bad news: I still get the error with the wrapped `UIManager#createLookAndFeel`:

```
18206: Error: java.lang.reflect.InvocationTargetException 
18206: Error:   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) 
18206: Error:   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:78) 
18206: Error:   at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) 
18206: Error:   at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) 
18206: Error:   at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480) 
18206: Error:   at org.omegat.util.gui.UIDesignManager.setTheme(UIDesignManager.java:81) 
18206: Error:   at org.omegat.util.gui.UIDesignManager.initialize(UIDesignManager.java:100) 
18206: Error:   at org.omegat.core.Core.initializeGUI(Core.java:237) 
18206: Error:   at org.omegat.Main.runGUI(Main.java:314) 
18206: Error:   at org.omegat.Main.main(Main.java:186) 
18206: Error: Caused by: javax.swing.UnsupportedLookAndFeelException: com.apple.laf.AquaLookAndFeel 
18206: Error:   at java.desktop/javax.swing.UIManager.createLookAndFeel(UIManager.java:547) 
18206: Error:   at org.madlonkay.desktopsupport.impl.Java9DesktopSupportImpl.createLookAndFeel(Java9DesktopSupportImpl.java:300) 
18206: Error:   at org.omegat.gui.theme.DelegatingLookAndFeel.<init>(DelegatingLookAndFeel.java:67) 
18206: Error:   at org.omegat.gui.theme.DefaultFlatTheme.<init>(DefaultFlatTheme.java:62) 
18206: Error:   ... 10 more 
```

Reply:
Actually I got it: I misunderstood that `createLookAndFeel` takes the LAF name, not the classname.

I've published a new version of DesktopSupport and pushed some changes to use it. This works for me on Mac with Java 8 and Java 16.

I note that the Motif theme doesn't work in Java 16 with this approach, but the other built-in LAFs do (Metal, Nimbus).